backend/api/views.py

```diff
@@ -55,66 +60,80 @@ def post(self, request):
+                                           category=category,
+                                           source=source)
 
         return JsonResponse({'success': True, 'cashflow_id': cashflow.id, 'creation_date': cashflow.created},
                             status=status.HTTP_201_CREATED)
 
     def get(self, request, cashflow_id):
         try:
+            # FIXME: Verify the user may access this cashflow (i.e. manages the account of this cashflow)
             cashflow = Cashflow.objects.get(id=cashflow_id)
-        except:
-            return JsonResponse({'success': False}, status=status.HTTP_400_BAD_REQUEST)
+        except Exception as e:
+            # FIXME: Differentiate between ObjectDoesNotExist and a broad Exception
+            print(e.__cause__)
+            return JsonResponse({'success': False}, status=status.HTTP_404_NOT_FOUND)
         serializer = CashflowSerializer(cashflow, many=False)
         return Response(serializer.data, status=status.HTTP_200_OK)
 
     def delete(self, _, cashflow_id):
         try:
+            # FIXME: Verify the user may access this cashflow (i.e. manages the account of this cashflow)
             cashflow = Cashflow.objects.get(id=cashflow_id)
-        except:
-            return JsonResponse({'success': False}, status=status.HTTP_400_BAD_REQUEST)
+        except Exception as e:
+            # FIXME: Differentiate between ObjectDoesNotExist and a broad Exception
+            print(e.__cause__)
+            return JsonResponse({'success': False}, status=status.HTTP_404_NOT_FOUND)
         cashflow.delete()
         return JsonResponse({'success': True, 'cashflow_id': cashflow_id}, status=status.HTTP_200_OK)
 
     def put(self, request, cashflow_id):
         try:
+            # FIXME: Verify the user may access this cashflow (i.e. manages the account of this cashflow)
             cashflow = Cashflow.objects.get(id=cashflow_id)
 
             cashflow.overall_value = request.data['overallValue']
-            cashflow.category, _ = Category.objects.get_or_create(name=request.data['category'])
-            place = request.data['place']
-            cashflow.place, _ = Place.objects.get_or_create(address=place['address'], name=place['name'])
+            cashflow.category, _ = Category.objects.get_or_create(
+                name=request.data['category'], defaults={"account": cashflow.account})
+            source = self.get_source_from_request(cashflow.account, request)
+
+            cashflow.source = source
             cashflow.updated = datetime.now()
             cashflow_type = request.data['type']
-        except:
+        except AttributeError as e:
+            print(e.__cause__)
+            return JsonResponse({'success': False, 'message': 'Invalid source type'},
+                                status=status.HTTP_400_BAD_REQUEST)
+        except Exception as e:
+            # Fixme differentiate between other exception types
+            print(e.__cause__)
```

Comment:
do you want to tackle this in this PR?

Reply:
No I think this will be too much and should be done in a seperate ticket